ReactiveCocoa/Swift/Action.swift

```diff
@@ -191,13 +191,7 @@ public final class CocoaAction: NSObject {
 	/// object given to execute(), if it can be downcast successfully, or nil
 	/// otherwise.
 	public convenience init<Input: AnyObject, Output, Error>(_ action: Action<Input?, Output, Error>) {
-		self.init(action, { input in
-			if let input: AnyObject = input {
-				return input as? Input
-			} else {
-				return nil
-			}
-		})
+		self.init(action, { $0 as? Input })
```

Comment:
I think what @NachoSoto is asking is why `AnyObject? as? Input` would work (since that's what this statement seems to imply). I have to admit that I don't understand why it works either.